src/spot_trade.circom

```diff
@@ -140,18 +145,26 @@ template orderUpdater(orderLevels) {
 template SpotTrade(balanceLevels, orderLevels, accountLevels) {
     signal input enabled;
 
+    signal input order1_pos;
     signal input order1_id;
     signal input order1_tokensell;
     signal input order1_amountsell;
     signal input order1_tokenbuy;
     signal input order1_amountbuy;
+    signal input order2_pos;
     signal input order2_id;
     signal input order2_tokensell;
     signal input order2_amountsell;
     signal input order2_tokenbuy;
     signal input order2_amountbuy;
-    order1_tokensell === order2_tokenbuy;
-    order1_tokenbuy === order2_tokensell;
+    component check1 = ForceEqualIfEnabled();
+    check1.enabled <== enabled;
+    check1.in[0] <== order1_tokensell
+    check1.in[1] <== order2_tokenbuy;
+    component check2 = ForceEqualIfEnabled();
+    check2.enabled <== enabled;
+    check2.in[0] <== order1_tokenbuy;
+    check2.in[1] <== order2_tokensell;
```

Comment:
Adding `enable` makes sense. Though I was manually making sure these in the encoded txs so that we reduce constraints.

I am fine with both ways.

Reply:
99% of constraints are hashing.  Adding several constraints here  to make code more modular seems to be worth